engines/kingdom/kingdom.cpp

```diff
@@ -0,0 +1,1628 @@
+				_mouseValue = mouseMapMS[var2 + i]._mouseValue;
+				break;
+			}
+		}
+	} else {
+		int tmpVal = checkMouseMapAS();
+		if (tmpVal == -1) {
+			cursorTypeExit();
+			return;
+		} else {
+			_mouseValue = tmpVal;
+		}
+	}
+
+	switch(_mouseValue) {
+	case 0x18A:
+		if (_logic->_eye)
+			_mouseValue = !_asMode ? 0x43A : 0x43B;
+		else
+			_mouseValue = 0;
+		break;
+	case 0x18C:
+		if (_treeLeftSta == 1)
+			_mouseValue = 0x43D;
+		else if (_treeLeftSta == 3)
+			_mouseValue = 0x43F;
+		else if (_treeLeftSta == 0)
+			_mouseValue = 0;
+		else if (_treeLeftSta == 2 && _logic->_replay)
+			_mouseValue = 0x43E;
+		else
+			_mouseValue = 0;
+		break;
+	case 0x18D:
+		if (_treeRightSta == 1)
+			_mouseValue = _logic->_help ? 0x43C : 0;
+		if (_treeRightSta == 2)
+			_mouseValue = 0x440;
+		break;
+	case 0x24A:
+		// Restore game.
+		// No more check in ScummVM, we display the load screen
+		break;
+	case 0x407:
+		if (_logic->_statPlay == 182 && _logic->_nodes[18] < 9)
+			_mouseValue = 0;
+		break;
+	case 0x40D:
+		if (_logic->_nodes[28] == 1)
+			_mouseValue = 0;
+		break;
+	case 0x41F:
+		if (_logic->_nodes[31] == 0)
+			_mouseValue = 0;
+		break;
+	case 0x422:
+	case 0x425:
+		if (!_wizard)
+			_mouseValue = 0;
+		break;
+	case 0x428:
+		if (_logic->_nodeNum == 5 && _gameMode != 2 && _logic->_spell1)
+			_mouseValue = 0;
+		break;
+	case 0x42A:
+		if (_logic->_nodeNum == 5 && _gameMode != 2 && _logic->_spell2)
+			_mouseValue = 0;
+		break;
+	case 0x42B:
+		if (_logic->_nodeNum == 5 && _gameMode != 2 && _logic->_spell3)
+			_mouseValue = 0;
+		break;
+	case 0x445:
+		if (_logic->_statPlay == 161 && _logic->_nodes[16] == 0 && _wizard)
+			_mouseValue = 0x450;
+		break;
+	case 0x44F:
+		if (!_logic->_pouch)
+			_mouseValue = 0;
+		break;
+	case 0x457:
+		if (!_logic->_tideCntl)
+			_mouseValue = 0;
+		break;
+	}
+
+	_iconSelect = 9;
+	for (int var6 = 0; var6 < 8; var6++) {
+		if (_mouseValue == 181 + var6) {
+			int var2 = _logic->_nodeNum;
+			if (_tsIconOnly)
+				var2 = 79;
+			if (_logic->_nodeNum == 56 && _logic->_inventory[8] < 1 && _wizard)
+				var2 = 80;
+			int indx = _iconActTable[var2][var6];
+			if (_logic->_inventory[indx] != 0 && _logic->_nodes[28] != 1 && _logic->_nodes[67] != 1 && !_iconsClosed && !_itemInhibit) {
+				_mouseValue = indx + 0x428;
+				_iconSelect = var6;
+				break;
+			}
+			_mouseValue = 0;
+		}
+	}
+
+	if (_logic->_currMap == 11) {
+		if (_mouseValue > 0x427 && _mouseValue < 0x43A) {
+			if (_logic->_inventory[_mouseValue - 0x428] < 1)
+				_mouseValue = 0x241;
+		}
+	}
+	cursorTypeExit();
+}
+
+void KingdomGame::cursorTypeExit() {
+	if (_mouseValue >= 0x400)
+		_cursorDef = _cursorTable[_mouseValue - 0x400];
+	else 
+		_cursorDef = (_mouseValue != 0) ? 0x68 : 0x67;
+}
+
+int KingdomGame::checkMouseMapAS() {
+	if (isDemo()) {
+		for (int i = 0; i < 16; i++) {
+			if (_cursorX >= _mouseMapASDemo[_logic->_currMap][i]._minX && _cursorX < _mouseMapASDemo[_logic->_currMap][i]._maxX
```

Comment:
You could set a variable with the value of _mouseMapASDemo[_logic->_currMap][i], so that you can reuse it, and make this code more readable

Reply:
It's now using a Common::Rect().contains(Common::Point)